hal/src/hal-filesystem.ads

```diff
@@ -0,0 +1,145 @@
+with Interfaces; use Interfaces;
+
+package HAL.Filesystem is
+
+   subtype Pathname is String;
+
+   type File_Kind is (Regular_File, Directory);
+
+   type File_Mode is (Read_Only, Write_Only, Read_Write);
+
+   type Permission_Kind  is (Others_Execute, Others_Write, Others_Read,
```

Comment:
This is highly FS-specific. In general, and in the context of Bare Metal FS, I'm not sure we'll need that. Then if we really want full support for permissions, most modern FS also support advanced ACL that go far beyond this definition.

My take on this is that all we need is queries on whether the application has read or write access to a specific path (I don't see us using the 'execution' access).